project/stac_api/models.py

```diff
@@ -1,28 +1,307 @@
-    layer_id = models.TextField()
-    geocat_uuid = models.UUIDField()
-    goecat_title = models.TextField()
-    geodat_description = models.TextField()
-    gsd = models.TextField()
-    variant = models.TextField()
-    lang = models.TextField()  # TODO: what is its purpose?
-    bbox = models.PolygonField(
-        default=BBOX_CH, srid=2056
-    )  # TODO: directly a geometry for spatial queries? 3D?
-    temporal_interval = models.TimeField(
-    )  # TODO: time field - how many hours are in a (leap) year? textfield?
-    crs = models.URLField(default="http://www.opengis.net/def/crs/OGC/1.3/CRS84")
+    crs = models.URLField(default=["http://www.opengis.net/def/crs/OGC/1.3/CRS84"])  # [string]
+    created = models.DateTimeField(auto_now_add=True)  # datetime
+    updated = models.DateTimeField(auto_now=True)  # datetime
+    description = models.TextField()  # string  / intentionally TextField and
+    # not CharField to provide more space for the text.
+    # TODO: ""description" is required in radiantearth spec, not required in our spec
+
+    # temporal extent will be auto-populated on every item update inside this collection:
+    # start_date will be set to oldest date, end_date to most current date
+    start_date = models.DateTimeField(blank=True)  # will be automatically populated
+    end_date = models.DateTimeField(blank=True)  # will be automatically populated
+
+    # spatial extent (2D ord 3D):
+    # bbox: southwesterly most extent followed by all axes of the northeasterly
+    # most extent specified in Longitude/Latitude or Longitude/Latitude/Elevation
+    # based on WGS 84.
+    # Example that covers the whole Earth: [[-180.0, -90.0, 180.0, 90.0]].
+    # Example that covers the whole earth with a depth of 100 meters to a height
+    # of 150 meters: [[-180.0, -90.0, -100.0, 180.0, 90.0, 150.0]].
+
+    # after discussion with Chris and Tobias:
+    # bbox resp. spatial extent will be auto-generated on collection level everytime
+    # when an item inside the collection is updated.
+    # the bbox of the collection will be a an envelope ("Umhüllende") of all
+    # bboxes of the items inside the collection.
+    # furthermore GeoDjango and its functionality will be used for that.
+    # TODO: overwrite items save() function accordingly
+    # suggestions of fields to be auto-populated:
+    SW = ArrayField(models.FloatField(), blank=True)  # [Float], auto-populated
+    NE = ArrayField(models.FloatField(), blank=True)  # [Float], auto-populated
+
+    collection_name = models.CharField(unique=True, default="")  # string
+    # collection_name is what is simply only called "id" in here:
+    # http://ltboc.infra.bgdi.ch/static/products/data.geo.admin.ch/apitransactional.html#operation/createCollection
+    itemType = models.CharField(default="Feature")  # string
+
+    # The field formerly known as "Keywords" has been "sourced out" to a
+    # separate Keyword class with many-to-many relations to the Collection objects
```

Comment:
Imho it's clearer to define the manytomany field here:
```
class Keywords(models.Model):
    name = models.CharField(max_length=64)

class Collection(models.Model):
    ...
    keywords = models.ManyToMany(Keywords)
```
this would allow to reuse the keyword model for M2M in e.g. an Item model

Reply:
Good point, Thanks! I changed that